src/panels/lovelace/cards/hui-clock-card.ts

```diff
@@ -0,0 +1,250 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../../components/ha-alert";
+import "../../../components/ha-card";
+import type { HomeAssistant } from "../../../types";
+import type {
+  LovelaceCard,
+  LovelaceCardEditor,
+  LovelaceGridOptions,
+} from "../types";
+import type { ClockCardConfig } from "./types";
+import { TimeFormat } from "../../../data/translation";
+import { useAmPm } from "../../../common/datetime/use_am_pm";
+import { resolveTimeZone } from "../../../common/datetime/resolve-time-zone";
+
+const INTERVAL = 1000;
+
+@customElement("hui-clock-card")
+export class HuiClockCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-clock-card-editor");
+    return document.createElement("hui-clock-card-editor");
+  }
+
+  public static getStubConfig(): ClockCardConfig {
+    return {
+      type: "clock",
+      clock_size: "medium",
+      show_seconds: "show",
+      show_am_pm: "auto",
+      time_format: "12",
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: ClockCardConfig;
+
+  @state() private _time?: Intl.DateTimeFormat;
+
+  @state() private _timeHour?: string;
+
+  @state() private _timeMinute?: string;
+
+  @state() private _timeSecond?: string;
+
+  @state() private _timeAmPm?: string;
+
+  private _tickInterval?: undefined | number;
+
+  public setConfig(config: ClockCardConfig): void {
+    this._config = config;
+  }
+
+  public getCardSize(): number {
+    if (this._config?.clock_size === "small") return 1;
+    return 2;
+  }
+
+  public getGridOptions(): LovelaceGridOptions {
+    if (this._config?.clock_size === "small") {
+      return {
+        min_rows: 1,
+        rows: 1,
+        max_rows: 4,
+        min_columns: 4,
+        columns: 6,
+      };
+    }
+
+    if (this._config?.clock_size === "large") {
+      return {
+        min_rows: 2,
+        rows: 2,
+        max_rows: 4,
+        min_columns: 6,
+        columns: 6,
+      };
+    }
+
+    return {
+      min_rows: 1,
+      rows: 2,
+      max_rows: 4,
+      min_columns: 4,
+      columns: 6,
+    };
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._startTick();
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._stopTick();
+  }
+
+  private _startTick() {
+    this._tick();
+    this._tickInterval = window.setInterval(() => this._tick(), INTERVAL);
+  }
+
+  private _stopTick() {
+    if (this._tickInterval) {
+      clearInterval(this._tickInterval);
+      this._tickInterval = undefined;
+    }
+  }
+
+  private _tick() {
+    const locale = this.hass?.locale;
+
+    if (!locale || !this.hass) return;
+
+    this._time = new Intl.DateTimeFormat(locale.language, {
```

Comment:
Create 1 instance of `Intl.DateTimeFormat` that you reuse, and only recreate it if you config changed or of `locale` changed.